src/SharedUtilities/utils.jl

```diff
@@ -17,3 +19,156 @@ end
 Helper function used in computing tendencies of vertical diffusion terms.
 """
 to_scalar_coefs(vector_coefs) = map(vector_coef -> vector_coef.u₃, vector_coefs)
+
+"""
+   dss!(Y::ClimaCore.Fields.FieldVector, p::ClimaCore.Fields.FieldVector, T::FT)
+
+Computes the weighted direct stiffness summation and updates `Y` in place.
+In the case of a column domain, no dss operations are performed.
+"""
+function dss!(
+    Y::ClimaCore.Fields.FieldVector,
+    _::ClimaCore.Fields.FieldVector,
+    _::FT,
+) where {FT}
+    for key in propertynames(Y)
+        property = getproperty(Y, key)
+        dss_helper!(property, axes(property))
+    end
+end
+
+"""
+    dss_helper!(field_vec::ClimaCore.Fields.FieldVector, _)
+
+Method of `dss_helper!` which unpacks properties of Y when on a
+domain that is 2-dimensional in the horizontal.
+
+The assumption is that Y contains FieldVectors which themselves contain either
+FieldVectors or Fields, and that the final unpacked variable is a Field.
+This method is invoked when the current property itself contains additional
+property(ies).
+"""
+function dss_helper!(field_vec::ClimaCore.Fields.FieldVector, _)
+    for key in propertynames(field_vec)
+        property = getproperty(field_vec, key)
+        dss_helper!(property, axes(property))
+    end
+end
+
+"""
+    dss_helper!(
+        field::ClimaCore.Fields.Field,
+        domain::Union{
+            ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace,
+            ClimaCore.Spaces.AbstractSpectralElementSpace,
+        })
+
+Method of `dss_helper!` which performs dss on fields of Y when on a
+domain that is 2-dimensional in the horizontal.
+
+The assumption is that Y contains FieldVectors which themselves contain either
+FieldVectors or Fields, and that the final unpacked variable is a Field.
+This method is invoked when the element cannot be unpacked further.
+"""
+function dss_helper!(
+    field::ClimaCore.Fields.Field,
+    domain::Union{
+        ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace,
+        ClimaCore.Spaces.AbstractSpectralElementSpace,
+    },
+)
+    Spaces.weighted_dss!(field)
+end
+
+"""
```

Comment:
should we have unit tests for these different methods?

Reply:
That's probably a good idea. Is there a simple setup that will allow us to tell when dss is performed vs not performed?

Reply:
hm. I dont know, but we could make a field, and compute dss on it, and save that value. and then compare after running this with the saved value?

Reply:
Okay, I added a test for dss in this PR